enzy_htp/_interface/pymol_interface.py

```diff
@@ -101,6 +102,111 @@ def select_pymol_obj(self, pattern: str, pymol_obj_name: str,
 
         return sorted(result)
 
+    def point_mutate(self, pos_key: Tuple[str, int],
+                     target: str, pymol_obj_name: str,
+                     pymol_session: pymol2.PyMOL,
+                     debug: bool = False) -> None:
+        """
+        Performs a single point mutation on the target WT in the PyMOL session in-place.
+        Args:
+            pos_key: the chain id and residue index.
+            target: the target residue name (3 letters).
+            pymol_obj_name: the name of the target WT in PyMOL.
+            pymol_session: the target PyMOL session.
+            debug: prints all rotamer scores out for debugging purposes.
+        Returns:
+            None.
+        """
+
+        # load pymol wizard mutagenesis function
+        pymol_session.cmd.wizard("mutagenesis")
+        pymol_session.cmd.do("refresh_wizard")
+        
+        # select res idx to mutate to target
+        pymol_session.cmd.get_wizard().set_mode(target)
+        if pymol_obj_name == "":
+            pymol_session.cmd.get_wizard().do_select(f"{pos_key[0]}/{str(pos_key[1])}/")
+        else:
+            pymol_session.cmd.get_wizard().do_select(f"{pymol_obj_name}//{pos_key[0]}/{str(pos_key[1])}/")
+
+        # prints all rotamers and strains out; also saves each variation to a PDB file in scratch/.
+        # can use for debugging purposes
+        if debug:
+            for i in range(1, pymol_session.cmd.count_states() + 1):
+                pymol_session.cmd.wizard("mutagenesis")
+                pymol_session.cmd.do("refresh_wizard")
+                pymol_session.cmd.get_wizard().set_mode(target)
+                if pymol_obj_name == "":
+                    pymol_session.cmd.get_wizard().do_select(f"{pos_key[0]}/{str(pos_key[1])}/")
+                else:
+                    pymol_session.cmd.get_wizard().do_select(f"{pymol_obj_name}//{pos_key[0]}/{str(pos_key[1])}/")
+            
+                pymol_session.cmd.get_wizard().do_state(i)
+                pymol_session.cmd.frame(i)
+                pymol_session.cmd.get_wizard().apply()
+                self.export_pymol_obj(pymol_obj_name, pymol_session, rotamer=i)
+            return
+
+        pymol_session.cmd.get_wizard().apply()
+
+    def export_pymol_obj(self, pymol_obj_name: str, 
+                        pymol_session: pymol2.PyMOL,
+                        if_retain_order: bool = True,
+                        rotamer: int = 0) -> str:
```

Comment:
Since this function is very general to saving PDB. It should know nothing about doing mutations. An alternative design that decouples its dependence on mutation knowledge is to change "rotamer" to "tag" or "name_tag" and in point_mutate() we figure out what name tag is passed to this function. (it could be something like "_point_mutate_R154W_rot_1")